packages/tools/accessibility/src/HtmlTwin/htmlTwinHostComponent.tsx

```diff
@@ -283,13 +295,17 @@ export class HTMLTwinHostComponent extends React.Component<IHTMLTwinHostComponen
         const queue: Control[] = [...rootItems];
         for (let i: number = 0; i < queue.length; i++) {
             const curNode = queue[i];
-            if (!curNode.isVisible || (!this._options.addAllControls && curNode.name !== "root" && !curNode.accessibilityTag?.description)) {
+            if (!curNode.isVisible || (!this._options.addAllControls && !(curNode instanceof Container) && !curNode.accessibilityTag?.description)) {
                 continue;
             }
-            if (curNode instanceof Container && curNode.children.length !== 0 && !(curNode instanceof Button)) {
+            if (
+                curNode instanceof Container &&
+                !(curNode instanceof Button) &&
+                ((this._options.addAllControls && curNode.children.length !== 0) || this._hasChildrenWithA11yTag(curNode))
+            ) {

```

Comment:
i'm not the biggest fan of instanceof, but there might be no way around it here?

The only thing I can see as improvement here is to first check:

```
const isContainer = curNode instanceof Container;
```

and then use it in both places instead of calling the same instanceof twice

Reply:
@RaananW Hmmm the Control class has `getDescendants` so we could check for that instead? For the Button we could simply check getClassName

Reply:
if it is possible not to use instanceof - even better. just trying to minimize the calls as much as possible. Maybe i'm pushing against it a bit too much, as it is a language feature, i just prefer a different approach.